contracts/cw3-dao/src/state.rs

```diff
@@ -103,22 +103,19 @@ impl Proposal {
         self.status = self.current_status(block);
     }
 
-    /// Helper function to check if a certain vote count has reached threshold.
-    /// Only called from is_rejected and is_passed for no and yes votes
-    /// Handles the different threshold types accordingly.
-    /// This function returns true if and only if vote_count is greater than the threshold which
-    /// is calculated.
-    /// In the case where we use yes votes, this function will return true if and only if the
-    /// proposal will pass.
-    /// In the case where we use no votes, this function will return true if and only if the
-    /// proposal will be rejected regardless of other votes.
-    fn does_vote_count_reach_threshold(&self, vote_count: Uint128, block: &BlockInfo) -> bool {
+    /// Returns true if this proposal is sure to pass (even before expiration, if no future
+    /// sequence of possible votes could cause it to fail).
+    pub fn is_passed(&self, block: &BlockInfo) -> bool {
         match self.threshold {
             Threshold::AbsolutePercentage {
                 percentage: percentage_needed,
             } => {
-                vote_count
-                    >= votes_needed(self.total_weight - self.votes.abstain, percentage_needed)
+                let opinions = self.total_weight - self.votes.abstain;
+                // If all abstain we cannot pass the proposal
+                if opinions.is_zero() {
+                    return false;
+                }
+                self.votes.yes >= votes_needed(opinions, percentage_needed)
```

Comment:
!opinions.is_zero() && (self.votes.yes >= votes_needed(opinions, percentage_needed))

Reply:
Think I like having this broken out into two things. Makes it a little easier on the eyes. :) Possible that we save a branch with the single conditional, but we have much less gas efficient code elsewhere.

Reply:
I'm not the biggest fan here, it makes it harder to read imo?

Reply:
then it would be 

if opinions.is_zero() {
  false 
} else {
  self.votes.yes >= votes_neeeded(opinions, percentage_needed)
}

anyhow thats a nit (just feels weird we return for 1 and not another)